src/bomf/filter/__init__.py

```diff
@@ -99,3 +99,40 @@ async def apply(self, candidates: List[Candidate]) -> List[Candidate]:
         filtered_aggregates = await self._base_filter.apply(aggregates)
         self._logger.info("There are %i filtered aggregates left", len(filtered_aggregates))
         return [self.disaggregate(fa) for fa in filtered_aggregates]
+
+CandidateProperty = TypeVar("CandidateProperty")
+class HardcodedFilter(Filter,ABC, Generic[CandidateProperty]):
+    """
+    a harcoded filter filters on a hardcoded list of allowed/blocked values (formerly known as white- and blacklist)
+    """
+    def __init__(self, criteria_selector:Callable[[Filter], CandidateProperty], values:Set[CandidateProperty]):
+        """
+        instantiate by providing a criteria selector that returns a property on which we can filter and a set of values.
+        Whether the values are used as allowed or not allowed (block) depends on the inheriting class
+        """
+        super().__init__()
+        self._criteria_selector = criteria_selector
+        self._values = values
+
+class BlocklistFilter(HardcodedFilter):
+    """
+    remove those candidates whose property is in the provided blocklist
+    """
+
+    async def predicate(self, candidate: Candidate) -> bool:
+        candidate_property = self._criteria_selector(candidate)
+        result = candidate_property not in self._values
+        if result is False:
+            self._logger.debug("'%s' is in the blocklist", candidate_property)
+        return result
+class AllowlistFilter(HardcodedFilter):
```

Comment:
wofür brauchen wir an der stelle die whitelist? an sich soll man ja migrieren was da ist außer es wird explizit ausgeschlossen

Reply:
kann ja auch den umgekehrten Fall geben: Z.B. kannst du den Vertrags-Status check auch umformulieren in einen Allowlist/Whitelist-Check wobei der candidate der Contract und die CandidateProperty der Status ist